### PR TITLE
config: warn if ignore_environments is set, but environment isn't

### LINKED ISSUE #91: Airbrake being run on test and development environments
Hey, 

My team is experiencing a problem with the new version of Airbrake. We are receiving error messages from our development and test environments even though we have
 config.ignore_environments = %w(development test)
in our config block.

Please advise what we may have done wrong :).

Richard

### PR DESCRIPTION
**Pull request by kyrylo**
Fixes #91 (Airbrake being run on test and development environments)

This is not the first time someone is confused about this behaviour.

### CODE REVIEW

File: lib/airbrake-ruby/config.rb [hunk 1]
```diff
@@ -138,6 +138,11 @@ def valid?
     # @return [Boolean] true if the config ignores current environment, false
     #   otherwise
     def ignored_environment?
+      if ignore_environments.any? && environment.nil?
+        logger.warn("#{LOG_LABEL} the 'environment' option is not set, " \
+                    "'ignore_environments' has no effect")
```
Comment (shifi) [inline lib/airbrake-ruby/config.rb]: Can you add test for this scenario? Thanks.

Comment (kyrylo) [inline lib/airbrake-ruby/config.rb]: Sure, added tests.

**Comment by kyrylo** [comment]
PTAL

File: spec/config_spec.rb [hunk 2]
```diff
@@ -156,4 +156,41 @@
       end
     end
   end
+
+  describe "#ignored_environment?" do
+    describe "warnings" do
+      context "when 'ignore_environments' is set and 'environment' isn't" do
+        it "prints a warning" do
+          out = StringIO.new
+          config = described_class.new(logger: Logger.new(out))
```
Comment (shifi) [inline spec/config_spec.rb]: This two lines are the same for all tests here:
```ruby
out = StringIO.new
config = described_class.new(logger: Logger.new(out))
```
Should you move them to `let` block inside `describe "warning" do`?

**Comment by shifi** [comment]
LGTM

File: spec/config_spec.rb (hunk 2 above)
Comment (kyrylo) [inline spec/config_spec.rb]: That would work. Moved, thanks.

**Pull request merged**